sdk/search/azure-search-documents/src/test/java/com/azure/search/documents/DataSourceSyncTests.java

```diff
@@ -6,65 +6,57 @@
-import io.netty.handler.codec.http.HttpResponseStatus;
 import org.junit.jupiter.api.Test;
 
 import java.net.HttpURLConnection;
+import java.util.ArrayList;
 import java.util.Iterator;
-import java.util.function.BiConsumer;
-import java.util.function.BiFunction;
-import java.util.function.Function;
-import java.util.function.Supplier;
+import java.util.List;
 
+import static com.azure.search.documents.TestHelpers.BLOB_DATASOURCE_TEST_NAME;
+import static com.azure.search.documents.TestHelpers.assertHttpResponseException;
+import static com.azure.search.documents.TestHelpers.generateRequestOptions;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
 
-public class DataSourceSyncTests extends SearchServiceTestBase {
+public class DataSourceSyncTests extends SearchTestBase {
     private static final String FAKE_DESCRIPTION = "Some data source";
     private static final String FAKE_STORAGE_CONNECTION_STRING =
         "DefaultEndpointsProtocol=https;AccountName=NotaRealAccount;AccountKey=fake;";
     private static final String FAKE_COSMOS_CONNECTION_STRING =
         "AccountEndpoint=https://NotaRealAccount.documents.azure.com;AccountKey=fake;Database=someFakeDatabase";
 
+    private final List<String> dataSourcesToDelete = new ArrayList<>();
     private SearchServiceClient client;
 
-    // commonly used lambda definitions
-    private BiFunction<DataSource, AccessOptions, DataSource> createOrUpdateDataSourceFunc =
-        (DataSource ds, AccessOptions ac) ->
-            createOrUpdateDataSource(ds, ac.getOnlyIfUnchanged(), ac.getRequestOptions());
-
-    private Supplier<DataSource> newDataSourceFunc = () -> createTestBlobDataSource(null);
-
-    private Function<DataSource, DataSource> mutateDataSourceFunc = (DataSource ds) -> 
-        ds.setDescription("somethingnew");
-
-    private BiConsumer<DataSource, AccessOptions> deleteDataSourceFunc = (DataSource dataSource, AccessOptions ac) ->
-        client.deleteDataSourceWithResponse(dataSource,
-            ac.getOnlyIfUnchanged(), ac.getRequestOptions(), Context.NONE);
-
     @Override
     protected void beforeTest() {
         super.beforeTest();
         client = getSearchServiceClientBuilder().buildClient();
     }
 
-    private DataSource createOrUpdateDataSource(DataSource datasource, Boolean onlyIfUnchanged,
-        RequestOptions requestOptions) {
-        return client.createOrUpdateDataSourceWithResponse(datasource, onlyIfUnchanged, requestOptions, Context.NONE)
-            .getValue();
+    @Override
+    protected void afterTest() {
+        super.afterTest();
+        for (String dataSource : dataSourcesToDelete) {
```

Comment:
Same here.
I have concern which the list has more sources than expected and throw no resource error.

Reply:
The concern is that this list would contain more data sources than still existing after the test? Luckily it appears 404 is a valid response type, so it won't throw an exception.